code/modules/reagents/Chemistry-Reagents.dm

```diff
@@ -9677,26 +9677,47 @@ var/global/list/tonio_doesnt_remove=list("tonio", "blood")
 /datum/reagent/ectoplasm
 	name = "Ectoplasm"
 	id = ECTOPLASM
-	description = "Pure, distilled spooky"
+	description = "Pure, distilled spooky."
 	reagent_state = REAGENT_STATE_LIQUID
 	color = "#21d389b4"
 	density = 0.05
-
+	custom_metabolism = 0.01
+	var/spookvision = FALSE
+	
 /datum/reagent/ectoplasm/on_mob_life(var/mob/living/M)
 	if(..())
 		return 1
+
+	if(tick >= 30 && volume >= 1 && !spookvision) //ghostsight after 1m and having more than 1u inside
+		spookvision = TRUE
+		to_chat(M, "<span class='notice'>You start seeing through the veil!</span>")
+		M.see_invisible = SEE_INVISIBLE_OBSERVER
+		M.see_invisible_override = SEE_INVISIBLE_OBSERVER
+
+	if(spookvision && volume < 1)
```

Comment:
unsure if intended, but this lets you "keep" your ticks, meaning if you drop under 1u, you still have 100 ticks / 200 seconds / about 3.5 minutes to get more ectoplasm and you won't have to wait 60 seconds to get the see invisible bonus again

did you test this for unintended see invisible consequences like seeing pipes or people under invisibility cloaks? I can't remember what their invisibility threshold is

Reply:
I tested and couldn't see anything other than ghosts, and kinda intended yeah, so you can keep the spookvision going if you manage.